users/views.py

```diff
@@ -1,10 +1,57 @@
-from django.contrib.auth.forms import UserCreationForm
-from django.urls import reverse_lazy
-from django.views import generic
+# from django.contrib.auth.forms import UserCreationForm
+# from django.urls import reverse_lazy
+# from django.views import generic
+from rest_framework import viewsets, generics
+from .models import UserProfile
+from .serializers import UserSerializer, CreateUserSerializer, CustomAuthTokenSerializer
+from users.permisions import IsOwnerIsAdmin, IsOwner
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.authtoken.views import ObtainAuthToken
+from rest_framework.authtoken.models import Token
+from rest_framework.response import Response
 
 
-class SignUp(generic.CreateView):
-    form_class = UserCreationForm
-    success_url = reverse_lazy('login')
-    template_name = 'registration/signup.html'
+class UserViewSet(viewsets.ModelViewSet):
+    queryset = UserProfile.objects.all()
+    serializer_class = UserSerializer
+    permission_classes = (IsAuthenticated, IsOwnerIsAdmin)
+
+
+class UsersView(generics.CreateAPIView):
+    serializer_class = CreateUserSerializer
+    permission_classes = (IsAuthenticated, IsOwnerIsAdmin)
```

Comment:
The aim of this view is creating new user. We should not use `IsAuthenticated, IsOwnerIsAdmin` as permission_class.
Can a new person create a user ?